RFC-009-SWAP-ERC20-basic.md

```diff
@@ -0,0 +1,321 @@
+```
+
+To compile the contract code replace the placeholder values with the HTLC paramters at the following offsets:
+
+| Data                     | Position of first byte | Position of last byte | Length (bytes) |
+|:-------------------------|:-----------------------|:----------------------|:---------------|
+| `secret_hash`            | 53                     | 84                    | 32             |
+| `expiry`                 | 102                    | 105                   | 4              |
+| `redeem_identity`        | 157                    | 176                   | 20             |
+| `refund_identity`        | 224                    | 253                   | 20             |
+| `token_quantity`         | 261                    | 292                   | 32             |
+| `token_contract_address` | 307                    | 338                   | 32             |
+
+
+
+The contract emits two logs:
+
+| Topic        | keccack256(Topic)                                                  | Data     | Emitted When             |
+|:-------------|:-------------------------------------------------------------------|----------|:-------------------------|
+| `Redeemed()` | 0xb8cac300e37f03ad332e581dea21b2f0b84eaaadc184a295fef71e81f44a7413 | `secret` | The contract is redeemed |
+| `Refunded()` | 0x5d26862916391bf49478b2f5103b0720a842b45ef145a268f2cd1fb2aed55178 | -        | The contract is refunded |
+
+
+Implementations SHOULD use the following gas limits on the transactions related to the contract:
+
+| Transaction | Recommended gas limit |
+|:------------|-----------------------|
+| Deployment  | 167,800               |
+| Funding     | 100,000               |
+| Redeem      | 100,000               |
+| Refund      | 100,000               |
+
+
+## Execution Phase
+
+The following section describes how both parties should interact with the Ethereum blockchain during the [RFC003 execution phase](./RFC-003-SWAP-basic.md#execution-phase).
+
+Note that with the ERC20 HTLC there's a *funding* stage after the *deployment* stage.
+The funding stage MUST be completed before the redeemer tries to redeem.
+
+
+### Deployment
+
+At the start of the deployment stage, both parties compile the contract code as described in the previous section.
+We will call this value `contract_code`.
+
+To deploy the ERC20 HTLC, the *funder* must deploy the `contract_code`.
+They SHOULD do this by sending a contract deployment transaction to the relevant Ethereum blockchain with `contract_code` as the `data` of the transaction.
+
+
+To be notified of the deployment event, both parties MAY watch the blockchain for a transaction with the `contract_code` as the data.
+Upon observing the deployment transaction, both parties SHOULD record the address the contract was deployed to (referred to as `contract_address` from now on).
+
+
+### Funding
+
+After deploying the contract, the funder must transfer ownership of the `token_quantity` to the `contract_address`.
```

Comment:
:x: `contract_address` is now the 3rd way of referring to the token contract :)

I think this should be more consistent, especially when you are using backticks to emphasize the way it is spelled.

Reply:
contract_address is the HTLC contract address. But yeah let's make this explicit.